src/Components/Facet.tsx

```diff
@@ -83,25 +94,55 @@ export default class Facet extends React.Component<IFacetProps, {}> {
               </Box>
             ))}
           </FormGroup>
-          {this.state.canShowMoreValues && (
-            <Button
-              onClick={() => {
-                this.showMore();
-              }}
-            >
-              Show More
-            </Button>
-          )}
-          {this.state.canShowLessValues && (
-            <Button
-              onClick={() => {
-                this.showLess();
-              }}
-            >
-              Show Less
-            </Button>
-          )}
         </FormControl>
+        {this.state.canShowMoreValues && (
```

Comment:
The more I think about it, the more I think we should split up the facet search into its own React component. At the moment, it's a bit tricky even finding where the code begins for the facet search specifically.

We could just pass the headless facet as a prop to the facet search component, that way everything would still work the way we want it, it would just be a bit easier to follow (at least I think).

This is just speculation though. What does everyone else think? 🙂

Reply:
I think this is a good idea 😄 
It would make both the facet and the facetsearch components clearer 👍

Reply:
Okay, here's the thing- none of the FacetSearch controller stuff is exported globally (I'm 95% sure, although they did recently change their file structure but from the various index files I'm fairly certain). So the only way to access the appropriate facet search controller is through the facet controller, meaning it must be part of the facet component.

I was thinking we could pass the facet search controller as a prop from the facet to the facet search component, and then we'd have access, but I'm not sure how to do it when I can't type all my variables properly. Is there a way around this, or am I coming at this all wrong?

Reply:
oh I see...
And indeed [in atomic it's all in the same file](https://github.com/coveo/ui-kit/blob/master/packages/atomic/src/components/atomic-facet/atomic-facet.tsx). 
One nice thing I see in that atomic file btw is that they split things up so [the render function is quite clean](https://github.com/coveo/ui-kit/blob/master/packages/atomic/src/components/atomic-facet/atomic-facet.tsx#L158). I don't know if that's be possible with MaterialUI, sometimes it behaves strangely...